### PR TITLE
EXLM-1349 Social Share Block UE & FE Changes

### PR DESCRIPTION
**Pull request by mayukhs2904**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1349

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live
- After: https://feat-exlm-1349--exlm--adobe-experience-league.hlx.live

### CODE REVIEW

**Review by devisree29 — APPROVED** [review]

File: blocks/social-share-block/social-share-block.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+.social-share-name{
+    color: var(--non-spectrum-grey-updated);
+    font-size: var(--spectrum-body-size-xs);
+}
+
+.social-share-item{
+    display: flex;
+    align-items: center;
+    gap: 6px;
+}
```
Comment (noormd08) [inline blocks/social-share-block/social-share-block.css:26]: Please format the CSS. Add space before curly bracket {

File: blocks/social-share-block/social-share-block.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      <path id="Path_749663" data-name="Path 749663" d="M158.6,147.95h2.275v1.044h.033a2.493,2.493,0,0,1,2.245-1.233c2.4,0,2.847,1.581,2.847,3.637v4.187h-2.373v-3.713c0-.885-.015-2.025-1.233-2.025-1.235,0-1.423.965-1.423,1.961v3.776H158.6Z" transform="translate(-152.363 -141.952)" fill="#fff"/>
+    </g>
+  </svg>
+  `,
+    'X (formerly Twitter)': `<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+    <rect id="Canvas" width="16" height="16" fill="#1065dc" opacity="0"/>
+  </svg>
+  `,
+  };
+
```
Comment (noormd08) [inline blocks/social-share-block/social-share-block.js:46]: SVG to be added as HTML Tag
const icon = '<span class="icon icon-**name**"></span>';

**Review by noormd08 — CHANGES_REQUESTED** [review]
Please check the review comments

File: blocks/social-share-block/social-share-block.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+export default function decorate(block) {
+  const socialDiv = block.firstElementChild;
+  const socialNetworks = socialDiv.textContent.split(',').map((network) => network.trim());
+
+  block.textContent = '';
+
+  const socialData = [
+    {
+      id: 'Facebook',
+      value: 'facebook',
```
Comment (noormd08) [inline blocks/social-share-block/social-share-block.js:25]: Please keep this one as ID and above one as Label

File: blocks/social-share-block/social-share-block.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      </div>
+      <div class="social-share-view">
+        ${socialNetworks
+          .map(
+            (network) =>
+              `<a href="${getObjectById(socialData, network).url}" target="_blank">
+                <div class="social-share-item">
+                  <span class="icon icon-${getObjectById(socialData, network).icon}"></span>
+                  <span class="social-share-name">${placeholders[getObjectById(socialData, network).value]}</span>
+                </div>
```
Comment (noormd08) [inline blocks/social-share-block/social-share-block.js:56]: Can you simplify this to a single call for getObjectById?

File: blocks/social-share-block/social-share-block.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    <div class="social-share-block">
+      <div class="social-share-title">
+        ${placeholders.shareOnSocial}
+      </div>
+      <div class="social-share-view">
+        ${socialNetworks
+          .map(
+            (network) =>
+              `<a href="${getObjectById(socialData, network).url}" target="_blank">
+                <div class="social-share-item">
```
Comment (noormd08) [inline blocks/social-share-block/social-share-block.js:53]: Do we need div tag inside the anchor? Can't we directly use the class name in anchor

**Review by noormd08 — APPROVED** [review]

**Pull request merged**